src/features/board/Board.tsx

```diff
@@ -15,11 +42,23 @@ export function Board () {
             <button onClick={() => dispatch(attack([1, 0]))}> Dragon attack Mage</button>
             <button onClick={() => dispatch(attack([1, 1]))}> Dragon attack Vampire</button>
             <div>
-                { opponentCards.map((card, i) => <Card key={`${card}${i}`} cardProperties={card} />) }
+                { opponentCards.map((card, i) => (
+                    <div>
+                        <Card key={`${card}${i}`} cardProperties={card} />
+                    </div>
+                )) }
             </div>
             <hr/>
             <div>
-                { playerCards.map((card, i) => <Card key={`${card}${i}`} cardProperties={card} />) }
+                { playerCards.map((card, i) => (
+                    <div
+                        onClick={() => { handleSelectCard(i) }}
```

Comment:
Here you have the memory leak issue. You thought you had no choice because you pass a parameter to `handleSelectedCard`.
However there is a solution. (It is not dramatic if you do not want to implement it here, it is just FYI)
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md#protips